ProjectDplusDsTree.py

```diff
@@ -147,13 +155,29 @@
     if 'cand_type' in dataFramePrompt.columns: #if not filtered tree, select only prompt and not reflected
         dataFramePrompt = FilterBitDf(dataFramePrompt, 'cand_type', [bitSignal, bitPrompt], 'and')
         dataFramePrompt = FilterBitDf(dataFramePrompt, 'cand_type', [bitRefl], 'not')
+        if args.LctopKpireso == 1:
+            dataFramePrompt = FilterBitDf(dataFramePrompt, 'cand_type', [bitLcNonRes], 'and')
+        if args.LctopKpireso == 2:
+            dataFramePrompt = FilterBitDf(dataFramePrompt, 'cand_type', [bitLcKStar], 'and')
+        if args.LctopKpireso == 3:
+            dataFramePrompt = FilterBitDf(dataFramePrompt, 'cand_type', [bitLcDelta], 'and')
+        if args.LctopKpireso == 4:
+            dataFramePrompt = FilterBitDf(dataFramePrompt, 'cand_type', [bitLcLambda1520], 'and')
```

Comment:
You can rewrite this by making a list of the different bits and accessing the list with `args.LctopKpireso` to avoid the copy-paste

```suggestion
        if args.LctopKpireso in range(1, 5):
            dataFramePrompt = FilterBitDf(dataFramePrompt, 'cand_type', [bitsLcResonance[args.LctopKpireso]], 'and')
```

Reply:
Done